impala/_thrift_api.py

```diff
@@ -120,6 +123,43 @@ def get_socket(host, port, use_ssl, ca_cert):
     else:
         return TSocket(host, port)
 
+def get_http_transport(host, port, http_path, timeout=None, use_ssl=False,
+                       ca_cert=None, auth_mechanism='NOSASL', user=None,
+                       password=None):
+    # TODO: support timeout
+    if timeout is not None:
+        log.error('get_http_transport does not support a timeout')
+    if use_ssl:
+        url = 'https://%s:%s/%s' % (host, port, http_path)
+        log.debug('get_http_transport url=%s', url)
+        # TODO: Add server authentication with thrift 0.12
+        transport = THttpClient(url)
+    else:
+        url = 'http://%s:%s/%s' % (host, port, http_path)
+        log.debug('get_http_transport url=%s', url)
+        transport = THttpClient(url)
+
+    if auth_mechanism == 'NOSASL':
+        return transport
+
+    # Set defaults for PLAIN SASL / LDAP connections.
+    if auth_mechanism in ['LDAP', 'PLAIN']:
+        if user is None:
+            user = getpass.getuser()
+            log.debug('get_http_transport: user=%s', user)
+        if password is None:
+            if auth_mechanism == 'LDAP':
+                password = ''
+            else:
+                # PLAIN always requires a password for HS2.
+                password = 'password'
+            log.debug('get_http_transport: password=%s', password)
```

Comment:
It doesn't look like we use password, or am I missing something - it's all ldap_password below.

Reply:
Fixed this one